.github/workflows/smoke.yml

```
@@ -0,0 +1,112 @@
+# These jobs are smoke tests for platforms where we don't rund full tests.
+# They ensure that std_spec, compiler_spec and the compiler itself at least
+# compile for the target with --cross-compile. But the binaries are not linked
+# and executed. So this does not validate correct behaviour.
+#
+# The list of supported platforms is extracted from the lib_c bindings:
+#
+# ```terminal-session
+# $ find src/lib_c -maxdepth 1 -mindepth 1 -type d -printf '%P\n' | sort
+# aarch64-darwin
+# aarch64-linux-gnu
+# aarch64-linux-musl
+# arm-linux-gnueabihf
+# i386-linux-gnu
+# i386-linux-musl
+# x86_64-darwin
+# x86_64-dragonfly
+# x86_64-freebsd
+# x86_64-linux-gnu
+# x86_64-linux-musl
+# x86_64-netbsd
+# x86_64-openbsd
+# x86_64-windows-msvc
+# ```
+#
+# Platforms for which we currently run full tests are excluded from this workflow.
+name: Smoke tests
+
+on: [push, pull_request]
+
+env:
+  TRAVIS_OS_NAME: linux
+  ARCH: x86_64
+  ARCH_CMD: linux64
+
+jobs:
+  aarch64-darwin:
+    env:
+      TARGET: aarch64-darwin
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download Crystal source
+      uses: actions/checkout@v2
+
+    - name: Build smoke test
+      run: bin/ci with_build_env make smoke_test target=$TARGET
+
+  arm-linux-gnueabihf:
+    env:
+      TARGET: arm-linux-gnueabihf
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download Crystal source
+      uses: actions/checkout@v2
+
+    - name: Build smoke test
+      run: bin/ci with_build_env make smoke_test target=$TARGET
+
+  i386-linux-musl:
+    env:
+      TARGET: i386-linux-musl
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download Crystal source
+      uses: actions/checkout@v2
+
+    - name: Build smoke test
+      run: bin/ci with_build_env make smoke_test target=$TARGET
+
+  x86_64-dragonfly:
+    env:
+      TARGET: x86_64-dragonfly
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download Crystal source
+      uses: actions/checkout@v2
+
+    - name: Build smoke test
+      run: bin/ci with_build_env make smoke_test target=$TARGET
+
+  x86_64-freebsd:
+    env:
+      TARGET: x86_64-freebsd
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download Crystal source
+      uses: actions/checkout@v2
+
+    - name: Build smoke test
+      run: bin/ci with_build_env make smoke_test target=$TARGET
+
+  x86_64-netbsd:
+    env:
+      TARGET: x86_64-netbsd
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download Crystal source
+      uses: actions/checkout@v2
+
+    - name: Build smoke test
+      run: bin/ci with_build_env make smoke_test target=$TARGET
+
+  x86_64-openbsd:
+    env:
+      TARGET: x86_64-openbsd
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download Crystal source
+      uses: actions/checkout@v2
+
+    - name: Build smoke test
+      run: bin/ci with_build_env make smoke_test target=$TARGET
```

This could use a `matrix` to make it more DRY.

```suggestion
jobs:
  smoke-test:
    runs-on: ubuntu-latest

    strategy:
      fail-fast: false
      matrix:
        target:
          - aarch64-darwin
          - arm-linux-gnueabihf
          - i386-linux-musl
          - x86_64-dragonfly
          - x86_64-freebsd
          - x86_64-netbsd
          - x86_64-openbsd

    env:
      TARGET: ${{ matrix.target }}

    steps:
      - name: Download Crystal source
        uses: actions/checkout@v2

      - name: Run smoke test
        run: bin/ci with_build_env make smoke_test target=$TARGET
```

---

Good idea! I was actually trying to do something like that, but didn't think about matrix. 🤦‍♂️ Thanks.